pkg/scheduler/plugins/predicates/gpu.go

```diff
@@ -50,44 +50,30 @@ func checkNodeGPUNumberPredicate(pod *v1.Pod, nodeInfo *api.NodeInfo) (bool, err
 	return true, nil
 }
 
-// predicateGPU returns the available GPU ID
+// predicateGPUbyMemory returns the available GPU ID
 func predicateGPUbyMemory(pod *v1.Pod, node *api.NodeInfo) []int {
 	gpuRequest := api.GetGPUMemoryOfPod(pod)
 	allocatableGPUs := node.GetDevicesIdleGPUMemory()
 
 	var devIDs []int
 
 	for devID := 0; devID < len(allocatableGPUs); devID++ {
-		availableGPU, ok := allocatableGPUs[devID]
-		if ok {
-			if availableGPU >= gpuRequest {
-				devIDs = append(devIDs, devID)
-				return devIDs
-			}
+		if availableGPU, ok := allocatableGPUs[devID]; ok && availableGPU >= gpuRequest {
+			devIDs = append(devIDs, devID)
 		}
 	}
-
-	return nil
+	return devIDs
 }
 
 // predicateGPU returns the available GPU IDs
 func predicateGPUbyNumber(pod *v1.Pod, node *api.NodeInfo) []int {
 	gpuRequest := api.GetGPUNumberOfPod(pod)
 	allocatableGPUs := node.GetDevicesIdleGPUs()
 
-	var devIDs []int
-
 	if len(allocatableGPUs) < gpuRequest {
 		klog.Errorf("Not enough gpu cards")
 		return nil
 	}
 
-	for devID := 0; devID < len(allocatableGPUs); devID++ {
-		devIDs = append(devIDs, allocatableGPUs[devID])
-		if len(devIDs) == gpuRequest {
-			return devIDs
-		}
-	}
-
-	return nil
+	return allocatableGPUs[:gpuRequest]
```

Comment:
Another concern here is should we always return the idle gpus, what about including the allocable gpus.

Reply:
@Thor-wl Is this make sense? Actually I haven't seen the original design intension.